src/main/java/uk/gov/pay/products/service/PaymentCreator.java

```diff
@@ -70,16 +71,19 @@ public Payment doCreate(String productExternalId, Long priceOverride, String ref
         return linksDecorator.decorate(paymentEntity.toPayment());
     }
 
-    private TransactionalOperation<TransactionContext, PaymentEntity> beforePaymentCreation(String productExternalId, String reference) {
+    private TransactionalOperation<TransactionContext, PaymentEntity> beforePaymentCreation(String productExternalId, String userDefinedReference) {
         return context -> {
             logger.info("Creating a new payment for product external id {}", productExternalId);
             ProductEntity productEntity = productDao.findByExternalId(productExternalId)
                     .orElseThrow(() -> new PaymentCreatorNotFoundException(productExternalId));
             if (productEntity.getReferenceEnabled()) {
-                if (isEmpty(reference)) {
+                if (isEmpty(userDefinedReference)) {
                     throw new BadPaymentRequestException("User defined reference is enabled but missing");
                 }
-                return mergePaymentEntityWithoutReferenceCheck(setupPaymentEntity(productEntity, reference));
+                if (PANDetector.isSuspectedPAN(userDefinedReference)) {
+                    logger.warn("Suspected PAN entered by user in reference field");
```

Comment:
Would it be useful to do this after we've created the payment so we can log a payment external id?

Reply:
Good point. I'd have to do it after the payment has been created, and also will lose context as to whether this was a user created reference, but that's fine